src/test/java/com/launchableinc/testng/TestSelectorTest.java

```diff
@@ -26,118 +29,109 @@ public void intercept_not_set_subset_list() {
-
-		TestNG tng = createTests("set-subset-list", Example1Test.class, Example2Test.class,
-				Example3Test.class);
-		tng.addListener(selector);
-		tng.run();
-		Assert.assertEquals(selector.totalTestCount, 9);
-		Assert.assertEquals(selector.filteredCount, 0);
+				"com.launchableinc.testng.Sample1Test\ncom.launchableinc.testng.Sample2Test\ncom.launchableinc.testok.Example1Test"
+						.getBytes(StandardCharsets.UTF_8));
+
+		assertNull(System.getenv(TestSelector.LAUNCHABLE_SUBSET_FILE));
+		withEnvironmentVariable(TestSelector.LAUNCHABLE_SUBSET_FILE, file.getPath()).execute(() -> {
+					TestSelector selector = new TestSelector();
+					TestNG tng =
+							createTests("set-subset-list", Example1Test.class, Example2Test.class,
+									Example3Test.class);
+					tng.addListener(selector);
+					tng.run();
+					assertEquals(selector.totalTestCount, 9);
+					assertEquals(selector.filteredCount, 9);
+		});
 
 		file.deleteOnExit();
 	}
 
 	@Test
-	public void intercept_set_not_exists_subset_file_path() throws IOException {
-		setEnv(TestSelector.LAUNCHABLE_SUBSET_FILE, "invalid_subset_file.txt");
-
-		TestSelector selector = new TestSelector();
-
-		TestNG tng = createTests("invalid_subset_file.txt", Example1Test.class, Example2Test.class,
-				Example3Test.class);
-		tng.addListener(selector);
-		tng.run();
-		// If subset file doesn't exist, selector will return the inputted methods as is.
-		Assert.assertEquals(selector.totalTestCount, 0);
-		Assert.assertEquals(selector.filteredCount, 0);
+	public void intercept_set_not_exists_subset_file_path() throws Exception {
+		assertNull(System.getenv(TestSelector.LAUNCHABLE_SUBSET_FILE));
+		withEnvironmentVariable(TestSelector.LAUNCHABLE_SUBSET_FILE, "invalid_subset_file.txt").execute(() -> {
+			TestSelector selector = new TestSelector();
+			TestNG tng = createTests("invalid_subset_file.txt", Example1Test.class, Example2Test.class,
+					Example3Test.class);
+			tng.addListener(selector);
+			tng.run();
+			// If subset file doesn't exist, selector will return the inputted methods as is.
+			assertEquals(selector.totalTestCount, 0);
+			assertEquals(selector.filteredCount, 0);
+		});
 	}
 
 	@Test
-	public void intercept_set_empty_subset_file() throws IOException {
+	public void intercept_set_empty_subset_file() throws Exception {
 		File file = File.createTempFile("subset-", ".txt");
-		Files.write(file.toPath(), "".getBytes(StandardCharsets.UTF_8));
-		setEnv(TestSelector.LAUNCHABLE_SUBSET_FILE, file.getPath());
-
-		TestSelector selector = new TestSelector();
-
-		TestNG tng = createTests("invalid_subset_file.txt", Example1Test.class, Example2Test.class,
-				Example3Test.class);
-		tng.addListener(selector);
-		tng.run();
-
-		Assert.assertEquals(selector.totalTestCount, 9);
-		Assert.assertEquals(selector.filteredCount, 0);
-
 		file.deleteOnExit();
-	}
+		Files.write(file.toPath(), "".getBytes(StandardCharsets.UTF_8));
 
-	private void setEnv(String key, String value) {
```

Comment:
I realized this method keeps env value untll remove or overwrite, and if we run tests in parallel, there might conflict it. So I introduce withEnvironmentVariable